index.js

```diff
@@ -189,46 +199,95 @@ class InflatingTransform extends Transform {
 			return callback(e)
 		}
 
-		this._push(generator, callback)
+		this._resumePushing(() => this._pushNextValue(generator, callback))
 	}
 
 	/**
-	 * Pushes values from a generator to the Readable stream.
+	 * Resumes pushing from a continuation.
 	 *
-	 * Handles any errors thrown when the generator yields a value.
+	 * Continuations are used to encapsulate execution state about what to "do next" as data is
+	 * pushed through the stream. This allows synchronous and asynchronous work
+	 * (both with promises and events) to be done by the stream as the "next step" for pushing a
+	 * chunk of data is encapsulated into a function and executed when required. Being a
+	 * function, the execution state can be passed to and returned from other functions (methods).
+	 *
+	 * Uses trampolining to avoid stack overflow with continuations.
+	 *
+	 * @see https://en.wikipedia.org/wiki/Continuation
+	 * @param {NextFunction} next What to do next
+	 * @return null If method is used in a continuation, signal no more work to be done.
+	 * @private
+	 */
+	_resumePushing(next) {
```

Comment:
As it stands, this method is pretty useless. It just wraps `trampoline`. I'd probably ditch it.

Reply:
But `_resumePushing` actually doesn't. It doesn't encapsulate any logic for resuming anything, or for pushing anything. The caller has to provide that. You could equally use it to generate a factorial....